packages/tenant-process/src/services/tenantService.ts

```diff
@@ -365,4 +460,95 @@ export function tenantServiceBuilder(
     },
   };
 }
+
+async function assignCertifiedAttribute({
+  targetTenant,
+  attribute,
+}: {
+  targetTenant: Tenant;
+  attribute: Attribute;
+}): Promise<Tenant> {
+  const certifiedTenantAttribute = targetTenant.attributes.find(
+    (attr): attr is CertifiedTenantAttribute =>
+      attr.type === tenantAttributeType.CERTIFIED && attr.id === attribute.id
+  );
+
+  // eslint-disable-next-line functional/no-let
+  let updatedTenant: Tenant = {
+    ...targetTenant,
+    updatedAt: new Date(),
+  };
+
+  if (!certifiedTenantAttribute) {
+    // assigning attribute for the first time
+    updatedTenant = {
+      ...updatedTenant,
+      attributes: [
+        ...targetTenant.attributes,
+        {
+          id: attribute.id,
+          type: tenantAttributeType.CERTIFIED,
+          assignmentTimestamp: new Date(),
+          revocationTimestamp: undefined,
+        },
+      ],
+    };
+  } else if (!certifiedTenantAttribute.revocationTimestamp) {
+    throw certifiedAttributeAlreadyAssigned(attribute.id, targetTenant.id);
+  } else {
+    // re-assigning attribute if it was revoked
+    updatedTenant = {
+      ...updatedTenant,
+      attributes: targetTenant.attributes.map((a) =>
+        a.id === attribute.id
+          ? {
+              ...a,
+              assignmentTimestamp: new Date(),
+              revocationTimestamp: undefined,
+            }
+          : a
+      ),
+    };
+  }
+  return updatedTenant;
+}
+
+async function revaluateTenantKind({
+  tenant,
+  version,
+  readModelService,
+  dbInstance,
+  correlationId,
+}: {
+  tenant: Tenant;
+  version: number;
+  readModelService: ReadModelService;
+  dbInstance: DB;
+  correlationId: string;
+}): Promise<Tenant> {
+  const tenantKind = await getTenantKindLoadingCertifiedAttributes(
+    readModelService,
+    tenant.attributes,
+    tenant.externalId
+  );
+
+  // eslint-disable-next-line functional/no-let
+  let updatedTenant: Tenant = { ...tenant };
+  if (tenant.kind !== tenantKind) {
+    updatedTenant = {
+      ...tenant,
+      kind: tenantKind,
+    };
+  }
```

Comment:
🧽 `let`
(the result is the same)
```suggestion
  const updatedTenant: Tenant = { 
    ...tenant,
    kind: tenantKind
  };
  if (tenant.kind !== tenantKind) {
    // create event here (but do not store it yet)
```